packages/native/src/components/Button/getButtonStyle.ts

```diff
@@ -1,75 +1,140 @@
-/* eslint-disable @typescript-eslint/no-unsafe-assignment */
-/* eslint-disable @typescript-eslint/no-unsafe-member-access */
 import { Theme } from "@ui/styles/theme";
+import { ButtonProps } from "@components/Button/index";
+import { TextTypes } from "@components/Text/getTextStyle";
 
-export type ButtonTypes = "primary" | "secondary";
-
-export default function getButtonStyle(p: {
-  type?: ButtonTypes;
-  disabled?: boolean;
-  theme: Theme;
-}): {
-  backgroundColor?: string;
-  borderColor?: string;
-  padding?: string;
+function getButtonColors(
+  colors: Theme["colors"],
+  props: ButtonProps
+): {
+  primaryColor: string;
+  secondaryColor: string;
+  tertiaryColor?: string;
 } {
-  switch (p.type) {
-    case "primary":
-      return p.disabled
-        ? {
-            backgroundColor: p.theme.colors.palette.neutral.c40,
-            padding: `0 ${p.theme.space[4]}px`,
-          }
-        : {
-            backgroundColor: p.theme.colors.palette.primary.c100,
-            padding: `0 ${p.theme.space[4]}px`,
-          };
-    case "secondary":
-      return p.disabled
-        ? {
-            borderColor: p.theme.colors.palette.neutral.c40,
-            padding: `0 ${p.theme.space[4]}px`,
-          }
-        : {
-            borderColor: p.theme.colors.palette.neutral.c40,
-            padding: `0 ${p.theme.space[4]}px`,
-          };
+  const { type, disabled, pressed, outline } = props;
+
+  const { palette } = colors;
+
+  const primaryColor = palette.neutral.c00;
+  let secondaryColor;
+  let tertiaryColor;
+
+  const disabledColors = {
+    primaryColor: palette.neutral.c50,
+    secondaryColor: outline ? palette.neutral.c50 : palette.neutral.c30,
+  };
+
+  switch (type) {
+    case "main":
+      if (disabled) {
+        return disabledColors;
+      } else if (pressed && !outline) {
```

Comment:
i would suggest in this case since you have multiple conditions to get one style to either so a switch(true) pattern
or a index one with { main: { disabled: {...}, outline {...} }

Also the pressed state doesnt make much sense in react native since you control that with the TouchableOpacity activeOpacity :grin:

Reply:
Pressed parameter removed (sorry for the Figma), now using only the opacity (it does the job)